tools/lsp/preview.rs

```diff
@@ -219,11 +219,20 @@ pub fn convert_diagnostics(
     diagnostics: &[slint_interpreter::Diagnostic],
 ) -> HashMap<lsp_types::Url, Vec<lsp_types::Diagnostic>> {
     let mut result: HashMap<lsp_types::Url, Vec<lsp_types::Diagnostic>> = Default::default();
+    let empty = PathBuf::new();
     for d in diagnostics {
-        if d.source_file().map_or(true, |f| f.is_relative()) {
+        let file_url = {
+            let file_path = d.source_file().unwrap_or(&empty).to_string_lossy().to_string();
+            if !file_path.contains(":/") && !file_path.is_empty() {
+                format!("file://{}", file_path)
```

Comment:
what if it was a relative file path?  there was a test before, but now it's gone.

Reply:
> The problem is: I can not even test for absolute paths in WASM :-/

you can add a `.starts_with("/")` for wasm.
But note that on the vscode web, all absolute path are going to be urls  (with `file://`  for vscode.dev when opening files, so that's why the file:// must stay.  Or some other random scheme)